fakeredis/_server.py

```diff
@@ -2655,39 +2654,10 @@ def check_is_ready_for_command(self, timeout):
 class FakeConnection(redis.Connection):
     description_format = "FakeConnection<db=%(db)s>"
 
-    def __init__(self, server, db=0, username=None, password=None,
-                 socket_timeout=None, socket_connect_timeout=None,
-                 socket_keepalive=False, socket_keepalive_options=None,
-                 socket_type=0, retry_on_timeout=False,
-                 encoding='utf-8', encoding_errors='strict',
-                 decode_responses=False, parser_class=_DummyParser,
-                 socket_read_size=65536, health_check_interval=0,
-                 client_name=None):
-        self.pid = os.getpid()
-        self.db = db
-        self.username = username
-        self.client_name = client_name
-        self.password = password
-        # Allow socket attributes to be passed in and saved even if they aren't used
-        self.socket_timeout = socket_timeout
-        self.socket_connect_timeout = socket_connect_timeout or socket_timeout
-        self.socket_keepalive = socket_keepalive
-        self.socket_keepalive_options = socket_keepalive_options or {}
-        self.socket_type = socket_type
-        self.retry_on_timeout = retry_on_timeout
-        self.encoder = redis.connection.Encoder(encoding, encoding_errors, decode_responses)
-        self._description_args = {'db': self.db}
-        self._connect_callbacks = []
-        self._buffer_cutoff = 6000
-        self._server = server
-        # self._parser isn't used for anything, but some of the
-        # base class methods depend on it and it's easier not to
-        # override them.
-        self._parser = parser_class(socket_read_size=socket_read_size)
-        self._sock = None
-        # added in redis==3.3.0
-        self.health_check_interval = health_check_interval
-        self.next_health_check = 0
+    def __init__(self, *args, **kwargs):
+        self._server = kwargs.pop('server')
+        self._description_args = {'db': kwargs.get('db', 0)}
```

Comment:
Is this still necessary? It was there for older versions of redis (e.g. 2.10.6), but `super().__init__` should now take care of it (and in fact is probably overwriting it).

Reply:
I see that the `FakeRedisMixin` constructor still accepts `server` parameter which is passed as a keyword to the Redis `ConnectionPool` class. In turn the `ConnectionPool` kwargs are passed to the constructor of the specific connection class.

Indeed the default Redis `Connection` class do not expect that parameter anymore in latest versions, but it seems it's needed by the `FakeConnection` class, precisely in the internal `_connect` method:

```python
def _connect(self):
        if not self._server.connected:
            raise redis.ConnectionError(CONNECTION_ERROR_MSG)
```

Regarding the `_description_args`, instead, I don't find any reference elsewhere. So it's probably a leftover.

Reply:
I was referring specifically to `_description_args`. Let's delete it.